### ISSUE TITLE
Conditional access support in IIdentity

### ISSUE THREAD

**Issue by vanillajonathan** [dotnet/runtime#25928]
This is not a feature request, more raising an issue for possible consideration.

When logging in from a new IP address, a new device or other criteria a identity provider could flag the authentication as limited.

The `IIdentity` interface could be extended with a property to communicate this information. Perhaps a boolean, possibly named `IsAuthenticationLimited` or something.

## Rationale and Usage
Allows you to hide/reduce functionality in case of any uncertainty in the authentication.
```cshtml
@if (User.Identity.IsAuthenticationLimited)
{
    <video src="movie_720p.webm" />
}
else
{
    <video src="movie_4k.webm" />
}
```

```cshtml
@if (!User.Identity.IsAuthenticationLimited)
{
    <button class="btn btn-danger" type="submit">Delete data and close my account</button>
}
```

```cs
[HttpPost]
public IActionResult DeleteAccount()
{
    if (User.Identity.IsAuthenticationLimited)
    {
        return BadRequest();
    }
    await _userManager.DeleteAsync(User);
    return View();
}
```

## Proposed API
```cs
    public interface IIdentity
    {
        string Name { get; }
        string AuthenticationType { get; }
        bool IsAuthenticated { get; }
        bool IsAuthenticationLimited { get; } // The possible addition
    }
```

## Open Questions
* Does any identity provider communicate this?
* Is this standardized in any authentication protocol such as OAuth or OpenID Connect?

**Comment by bartonjs** [comment]
Even though Default Interface Methods have been added to the C# language and the .NET Runtime, our current breaking change guidelines don't allow altering an interface once it releases, even via DIMs.

Personally, I'm not sure what "limited" means in this context.  So I don't really have an alternative suggestion, other than saying it might be an app-specific problem that requires using a derived interface and as-casts.

```C#
internal static class IdentityExtensions
{
    internal static bool IsAuthenticationLimited(this IIdentity identity)
    {
        if (identity == null)
            throw new ArgumentNullException(nameof(identity));

        if (identity is ILimitableIdentity limitable)
            return limitable.IsAuthenticationLimited;

        // Probably false is the right default?
        return false;
    }
}
```

**Comment by vanillajonathan** [comment]
Yeah, it might be a bit of a app-specific problem. I like your idea of using a extension method.

Limited could mean:
- User logged in from a new / previously unknown IP address or location.
- User logged in without two-factor authentication.
- User logged in remotely, not from the intranet.
- User logged in from a new device.
- User logged in without a client certificate.
- User logged in from a device which did not have a verified/trusted/secure boot.

**Comment by KalleOlaviNiemitalo** [comment]
[JWT claims](https://www.iana.org/assignments/jwt/jwt.xhtml#claims) "acr" (Authentication Context Class Reference), "amr" (Authentication Methods References, see [registry](https://www.iana.org/assignments/authentication-method-reference-values/authentication-method-reference-values.xhtml#amr-values)), and "vot" (Vector of Trust value, see [registry](https://www.iana.org/assignments/vot/vot.xhtml#vot-components)) may be relevant.
